### ISSUE TITLE
Jekyll-Archives (tags, categories) don't work

### ISSUE THREAD

**Issue by USTCqzy** [alshedivat/al-folio#1024]
**Describe the bug**
My homepage deploys al-folio, but all categories and part of tags don't work.

**To Reproduce**
1. [this tag](https://ustcqzy.github.io/blog/tag/%E7%BB%8F%E9%AA%8C%E5%88%86%E4%BA%AB/) works
2. Any category, e.g. [this category](https://ustcqzy.github.io/blog/category/PaperReading/) doesn't work

**Expected behavior**
Fix to see features in sample

**Screenshots**
N/A

**System (please complete the following information):**
 - OS: macOS
 - Browser (and its version) neither chrome nor safari works
 - Jekyll version 4.3.1
- Ruby version 3.0.3p157

**Additional context**
N/A

**Comment by alshedivat** [comment]
@AbstractGeek, do you happen to know why some tags or categories may not be working?

**Comment by AbstractGeek** [comment]
I think Jekyll makes tags and categories all lowercase. The same category that did not work, works when everything is made into a lowercase - https://ustcqzy.github.io/blog/category/paperreading/ (instead of https://ustcqzy.github.io/blog/category/PaperReading/). 

@USTCqzy An easy fix is to make all tags and categories lower case for now. That will make sure there won't be any broken links.

@alshedivat The fix should be simple. We just have to lowercase while making the links on the blog page (and wherever tags and categories are generated). Do you want me to make a pull request for it, if you are not in a hurry (currently tied up with a few things)?

**Comment by alshedivat** [comment]
@AbstractGeek, thanks for looking into this. no rush and would appreciate it if you could send a PR with a fix!

**Comment by tostenzel** [comment]
I have a similar problem. All my tags work locally but not remote (https://www.tobiasstenzel.com/blog/)

**Comment by george-gca** [comment]
How do you upload your site to your custom domain? I am thinking of now, maybe it is a permissions issue? The file is there, but the server doesn't have read permission?

**Comment by tostenzel** [comment]
I followed this recipe https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site.

As I understand it, www.tobiasstenzel.com mirrors tostenzel.github.io.
And the setting on the github pages repo is such that I am automatically re.directed to the .com domain. I think there should not be any differences. The setting for at my DNS provider site is also simply a CNAME entry -- no extra magic that could go wrong there, I guess...

**Comment by TracyYXChen** [comment]
I encountered the same problem as @ziyueqiu : Chinese tags work, but not English ones. I managed to fix them by doing the following:
1. make sure tags are all lower-case, and do not contain empty spaces
2. in `_config.yml`, make sure the tag names are all listed in `display_tags` 
3. in each post's markdown file, include the tag names in `tags: `, not `categories`

but why do we need both `tags` and `categories`? If there is no way to display all categories, i.e., there is no `display_categories` entry in `_config.yml`
